### PR TITLE
Delete reference to simpleTypeSpecifier

### PR DESCRIPTION
**Pull request by meg-gupta**
Update as per swift-syntax changes in https://github.com/swiftlang/swift-syntax/pull/2876

### CODE REVIEW

**Comment by stmontgomery** [comment]
@swift-ci please test

**Comment by stmontgomery** [comment]
It looks like this will break compatibility with the swift-syntax-600.0.x tag that the package depends on. Is there a way we can stage this change such that it will remain compatible with earlier tags of swift-syntax? Alternatively, could we first get this change included in a swift-syntax tag, then update the package dependency to reference that tag and then land this change?

**Comment by grynspan** [comment]
Should be possible to do something like:

```swift
#if !canImport(SwiftSyntax, _version: N)
// old code here
#endif
```

Where `N` is whatever tag of swift-syntax deleted this symbol.

File: Sources/TestingMacros/TestDeclarationMacro.swift [hunk 1]
```diff
@@ -86,9 +86,6 @@ public struct TestDeclarationMacro: PeerMacro, Sendable {
       let invalidSpecifierKeywords: [TokenKind] = [.keyword(.inout), .keyword(.isolated), .keyword(._const),]
       if let parameterType = parameter.type.as(AttributedTypeSyntax.self) {
         for specifier in parameterType.specifiers {
-          guard case let .simpleTypeSpecifier(specifier) = specifier else {
-            continue
-          }
           if invalidSpecifierKeywords.contains(specifier.specifier.tokenKind) {
```
Comment (grynspan) [inline Sources/TestingMacros/TestDeclarationMacro.swift:89]:
```suggestion
          for token in specifier.tokens(viewMode: .sourceAccurate) {
            if invalidSpecifierKeywords.contains(token.tokenKind) {
              diagnostics.append(.specifierNotSupported(token, on: parameter, whenUsing: testAttribute))
              break
            }
          }
```

**Comment by meg-gupta** [comment]
@swift-ci please test

**Comment by grynspan** [comment]
Based on a side discussion, I think we can close this PR?

**Comment by meg-gupta** [comment]
Closing this PR. Reworked https://github.com/swiftlang/swift-syntax/pull/2876 to avoid api break

**Pull request closed without merging**